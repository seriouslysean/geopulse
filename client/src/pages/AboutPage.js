import React from 'react';

export const AboutPage = () => (
  <div className="page-about">
    <div className="page-content">
      <section className="section section--spacing section--bg-sky-light">
        <h1 className="section__title">About</h1>
        <div className="content-container">
          <div className="content">
            <p>
              Geopulse is a tool that allows you to see content based on your location including a
              Google Map of the area, the local weather from Weather Underground, photos from
              Instagram, tweets from Twitter and videos from YouTube.
            </p>
          </div>
        </div>
      </section>
      <section className="section section--spacing section--bg-earth-light">
        <h1 className="section__title">Privacy</h1>
        <div className="content-container">
          <div className="content">
            <p>
              No data is saved on our servers and is only requested on an as-needed basis. Your
              personal information (ip address, etc) is not sold, rented or shared with a third
              party, unless required by law.
            </p>
            <p>
              Anonymous information is tracked via{' '}
              <a rel="external" href="https://www.google.com/analytics/" title="Google Analytics">
                Google Analytics
              </a>{' '}
              (which uses cookies) while you browse the site.
            </p>
          </div>
        </div>
      </section>
      <section className="section section--spacing section--bg-earth-medium technology">
        <h1 className="section__title">Technology</h1>
        <div className="content-container">
          <div className="content">
            <p>The following technologies were used in the creation of Geopulse.</p>
            <ul className="technology__list">
              <li>
                <a rel="external" href="https://nodejs.org/en/" title="Node.js" target="_blank">
                  Node.js
                </a>
              </li>
              <li>
                <a rel="external" href="https://expressjs.com/" title="Express" target="_blank">
                  Express
                </a>
              </li>
              <li>
                <a rel="external" href="https://ifconfig.io/" title="ifconfig.io" target="_blank">
                  ifconfig.io
                </a>
              </li>
              <li>
                <a
                  rel="external noopener noreferrer"
                  href="https://freegeoip.net/"
                  title="freegeoip.net"
                  target="_blank"
                >
                  freegeoip.net
                </a>
              </li>
              <li>
                <a
                  rel="external noopener noreferrer"
                  href="https://developers.google.com/maps/"
                  title="Google Maps"
                  target="_blank"
                >
                  Google Maps
                </a>
              </li>
              <li>
                <a
                  rel="external noopener noreferrer"
                  href="https://www.wunderground.com/weather/api/"
                  title="Weather Underground"
                  target="_blank"
                >
                  Weather Underground
                </a>
              </li>
              <li>
                <a
                  rel="external noopener noreferrer"
                  href="https://www.instagram.com/developer/"
                  title="Instagram"
                  target="_blank"
                >
                  Instagram
                </a>
              </li>
              <li>
                <a
                  rel="external noopener noreferrer"
                  href="https://developer.twitter.com/"
                  title="Twitter"
                  target="_blank"
                >
                  Twitter
                </a>
              </li>
              <li>
                <a
                  rel="external noopener noreferrer"
                  href="https://developers.google.com/youtube/"
                  title="YouTube"
                  target="_blank"
                >
                  YouTube
                </a>
              </li>
              <li>
                <a rel="external" href="https://www.heroku.com/" title="Heroku" target="_blank">
                  Heroku
                </a>
              </li>
              <li>
                <a
                  rel="external noopener noreferrer"
                  href="https://www.cloudflare.com/"
                  title="Cloudflare"
                  target="_blank"
                >
                  Cloudflare
                </a>
              </li>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </div>
);

export default {
  component: AboutPage,
};
