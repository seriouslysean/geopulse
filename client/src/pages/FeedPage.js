import React from "react";
import { connect } from "react-redux";
import config from "../../../config/config";
import { getGeolocation } from "../actions/geolocation";
import { getWeather } from "../actions/weather";
import { getPhotos } from "../actions/photos";
import { getChatter } from "../actions/chatter";
import withGeolocation from "../hocs/withGeolocation";
import withWeather from "../hocs/withWeather";
import withPhotos from "../hocs/withPhotos";
import withChatter from "../hocs/withChatter";
import Location from "../components/Location";
import Weather from "../components/Weather";
import Photos from "../components/Photos";
import Chatter from "../components/Chatter";
import Videos from "../components/Videos";

const LocationWithGeolocation = withGeolocation(Location);
const WeatherWithWeather = withWeather(Weather);
const PhotosWithPhotos = withPhotos(Photos);
const ChatterWithChatter = withChatter(Chatter);
const VideosWithVideos = withGeolocation(Videos);

class FeedPage extends React.Component {
  componentDidMount() {
    this.props.getGeolocation().then(location => {
      const { latitude, longitude } = location;
      this.props.getWeather(latitude, longitude);
      this.props.getPhotos(latitude, longitude);
      this.props.getChatter(latitude, longitude);
    });
  }
  render() {
    return (
      <div className="page-feed">
        <div className="page-feed">
          <LocationWithGeolocation
            googleMapURL={`https://maps.googleapis.com/maps/api/js?key=${
              config.GOOGLE_MAPS_API_KEY
            }&v=3.exp&libraries=geometry,places`}
            loadingElement={<div style={{ height: `100%` }} />}
            containerElement={<div style={{ height: `400px` }} />}
            mapElement={<div style={{ height: `100%` }} />}
          />
          <WeatherWithWeather />
          <PhotosWithPhotos />
          <ChatterWithChatter />
          <VideosWithVideos />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  getGeolocation: () => dispatch(getGeolocation()),
  getWeather: (latitude, longitude) => dispatch(getWeather(latitude, longitude)),
  getPhotos: (latitude, longitude) => dispatch(getPhotos(latitude, longitude)),
  getChatter: (latitude, longitude) => dispatch(getChatter(latitude, longitude))
});

export default {
  component: connect(undefined, mapDispatchToProps)(FeedPage)
};
