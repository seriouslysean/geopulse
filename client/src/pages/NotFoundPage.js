import React from "react";

const NotFoundPage = () => (
  <div className="page-not-found">
    <div className="page-content">
      <section className="section section--spacing section--bg-sky-light">
        <div className="content-container">
          <h1 className="section__title">Whoops!</h1>
          <div className="content">
            <p>Something isn't quite right here, we weren't able to find what you were looking for.</p>
          </div>
        </div>
      </section>
    </div>
  </div>
);

export default {
  component: NotFoundPage
};
