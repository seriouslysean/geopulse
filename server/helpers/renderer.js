import React from 'react';
import serialize from 'serialize-javascript';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import { Helmet } from 'react-helmet';
import AppRoutes from '../../client/src/routes/AppRoutes';

export default (req, store, context) => {
  const html = (
    <Provider store={store}>
      <StaticRouter location={req.path} context={context}>
        <div className="app-container">{renderRoutes(AppRoutes)}</div>
      </StaticRouter>
    </Provider>
  );
  const content = renderToString(html);
  const helmet = Helmet.renderStatic();

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />

        ${helmet.title.toString()}
        ${helmet.meta.toString()}

        <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
        <link rel="manifest" href="/manifest.json">
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#43578c">

        <!-- iOS -->
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <meta name="apple-mobile-web-app-title" content="Geopulse">
        <link rel="apple-touch-startup-image" href="/assets/geopulse-splash.png">
      
        <!-- Other -->
        <meta name="application-name" content="Geopulse">
        <meta name="theme-color" content="#43578c">
        <meta name="format-detection" content="telephone=no">
      
        <!-- Social -->
        <meta itemprop="name" content="Geopulse">
        <meta itemprop="description" content="Get a pulse on the happenings nearby">
        <meta itemprop="image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
      
        <meta property="og:image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
        <meta property="og:image:width" content="1200">
        <meta property="og:image:height" content="628">
        <meta property="og:title" content="Geopulse">
        <meta property="og:description" content="Get a pulse on the happenings nearby">
        <meta property="og:url" content="https://geopulse-app.herokuapp.com">
      
        <meta name="twitter:card" content="summary_large_image">
        <meta name="twitter:title" content="Geopulse">
        <meta name="twitter:description" content="Get a pulse on the happenings nearby">
        <meta name="twitter:creator" content="@seriouslysean">
        <meta name="twitter:image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
        
        <link rel="stylesheet" type="text/css" href="bundle.css" />
      </head>
      <body>
        <div id="root">${content}</div>
        <script>window.INITIAL_STATE = ${serialize(store.getState())}</script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;

  res.send(html);
};
