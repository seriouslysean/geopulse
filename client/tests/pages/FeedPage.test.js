import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { FeedPage } from "../../src/pages/FeedPage";

test("should render FeedPage correctly", () => {
  const getGeolocation = jest.fn().mockReturnValue({
    then: jest.fn().mockReturnValue({
      catch: jest.fn()
    })
  });
  const setGeolocationUnavailable = jest.fn();
  const getWeather = jest.fn();
  const getPhotos = jest.fn();
  const getChatter = jest.fn();
  const getVideos = jest.fn();

  const wrapper = shallow(
    <FeedPage
      getGeolocation={getGeolocation}
      setGeolocationUnavailable={setGeolocationUnavailable}
      getWeather={getWeather}
      getPhotos={getPhotos}
      getChatter={getChatter}
      getVideos={getVideos}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
