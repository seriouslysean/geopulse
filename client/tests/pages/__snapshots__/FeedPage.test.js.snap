// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FeedPage correctly 1`] = `
<div
  className="page-feed"
>
  <div
    className="page-content"
  >
    <div
      className="section-container"
    >
      <Connect(HOC)
        containerElement={
          <div
            style={
              Object {
                "height": "400px",
              }
            }
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?key=undefined&v=3.exp&libraries=geometry,places"
        loadingElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
      />
      <Connect(HOC) />
      <Connect(HOC) />
      <Connect(HOC) />
      <Connect(HOC) />
    </div>
  </div>
</div>
`;
